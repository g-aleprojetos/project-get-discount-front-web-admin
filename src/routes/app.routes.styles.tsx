import styled from 'styled-components';
import colors from 'resources/colors';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background-color: ${colors.lightgray};
`;
