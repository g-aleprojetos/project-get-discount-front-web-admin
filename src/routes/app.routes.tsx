import * as React from 'react';
import {Routes, Route} from 'react-router-dom';
import rotas from 'types/rotasAPP';
import {Roles} from 'types/enuns';
import {UserPage} from 'Pages/Userpage';
import {NotAuthorizedPage} from 'Pages/NotAuthorizedPage';
import {NotFoundPage} from 'Pages/NotFoundPage';
import {AdminPage} from 'Pages/AdminPage';
import {LoginPage} from 'Pages/Login';
import {PublicRoutes} from 'helpers/PublicRoutes';
import {PrivateRoutes} from 'helpers/PrivateRoutes';
import {useAuthenticatorContext} from 'context/authetication';
import {Header} from 'Components/header';
import * as S from './app.routes.styles';

export const WebRotas = () => {
  const {autenticado} = useAuthenticatorContext();
  return (
    <S.Container data-testid={'test_rotas'}>
      {autenticado && <Header />}
      <Routes>
        <Route path="/login" element={<PublicRoutes />}>
          <Route path={rotas.Login} element={<LoginPage />} />
        </Route>
        <Route
          path="/"
          element={<PrivateRoutes roles={[Roles.ADMIN, Roles.USER]} />}>
          <Route path={rotas.UserPage} element={<UserPage />} />
          <Route path={rotas.NaoAutorizado} element={<NotAuthorizedPage />} />
          <Route path={rotas.NotFound} element={<NotFoundPage />} />
          <Route path="/" element={<PrivateRoutes roles={[Roles.ADMIN]} />}>
            <Route path={rotas.AdminPage} element={<AdminPage />} />
          </Route>
        </Route>
      </Routes>
    </S.Container>
  );
};
