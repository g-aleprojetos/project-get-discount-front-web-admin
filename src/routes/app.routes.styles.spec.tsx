import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from './app.routes.styles';

describe('NotAuthorizedPage.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
