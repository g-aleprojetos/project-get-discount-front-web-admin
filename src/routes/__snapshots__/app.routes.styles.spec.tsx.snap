// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAuthorizedPage.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background-color: #D3D3d3;
}

<div
  className="c0"
/>
`;
